absl/hash/internal/hash.h

```diff
@@ -500,6 +501,26 @@ AbslHashValue(H hash_state, const std::vector<T, Allocator>& vector) {
                     vector.size());
 }
 
+#if defined(ABSL_IS_BIG_ENDIAN)
+// AbslHashValue for hashing std::vector<bool>
+//
+// std::hash does not work correctly with vector<bool> on Big Endian platforms
+// therefore we need to implement a custom AbslHashValue for it.
+// More details on the bug: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=102531
+template <typename H, typename T, typename Allocator>
+typename std::enable_if<is_hashable<T>::value && std::is_same<T, bool>::value,
+                        H>::type
+AbslHashValue(H hash_state, const std::vector<T, Allocator>& vector) {
+  std::vector<unsigned char> v;
```

Comment:
Actually, now that I think about it, we shouldn't use `std::vector` here because it since it allocates, it may throw.

I think you can probably use a stack array along with `PiecewiseCombiner` (see above or how it is used in `absl::Cord`) to hash this in pieces without allocating.

Reply:
Thanks for information, I have refactored it to use PiecewiseCombiner. Not sure what you meant by using a stack array as vector size is not a constant and can't use VLAs in C++, I'm simply passing the address of the input 1 byte at a time, please let me know if this approach is acceptable.